mne/decoding/time_frequency.py

```diff
@@ -0,0 +1,161 @@
+# Author: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from .mixin import TransformerMixin
+from .base import BaseEstimator
+from ..time_frequency.tfr import _compute_tfr, _check_tfr_param
+
+
+class TimeFrequency(TransformerMixin, BaseEstimator):
+    """Time frequency transformer.
+
+    Time-frequency transform of times series along the last axis.
+
+    Parameters
+    ----------
+    frequencies : array-like of floats, shape (n_freqs)
+        The frequencies.
+    sfreq : float | int, defaults to 1.0
+        Sampling frequency of the data.
+    method : 'multitaper' | 'morlet', defaults to 'morlet'
+        The time-frequency method. 'morlet' convolves a Morlet wavelet.
+        'multitaper' convolves DPSS tapers.
+    n_cycles : float | array of float, defaults to 7.0
+        Number of cycles  in the Morlet wavelet. Fixed number
+        or one per frequency.
+    zero_mean : bool | None, defaults to None
```

Comment:
for new object we should not expose this zero_mean parameter. Wavelets should be zero mean. The zero mean correction of Morlet is just not always applied in academic implementation (typically I had not done this early on and I had to add the zero_mean param and set it to False to avoid backward compat issues). So I would just not expose it here and make it True all the time.